pkg/cmd/adm/restart_test.go

```diff
@@ -1,216 +1,390 @@
+					require.NotNil(t, actual.Spec.Replicas)
+					assert.Equal(t, int32(1), *actual.Spec.Replicas)
+					require.NotNil(t, actual.Annotations["restartedAt"])
+				})
+				assert.Equal(t, 2, csCalls)
+				require.Contains(t, term.Output(), "Checking the status of the rolled out deployment")
+				require.Contains(t, term.Output(), "Running the Rollout status to check the status of the deployment")
+			} else if tc.labelValue == "codeready-toolchain" {
+				//Checking the logic where no operator deployments are there
+				require.Error(t, err, "no operator based deployment found in namespace toolchain-host-operator , hence no restart happened")
+				assert.Equal(t, 0, csCalls)
+			} else if tc.labelValue == "kubesaw-controller-manager" && tc.name1 != "autoscaling-buffer" {
+				//checking the logic when only operator based deployment is there and no non-operator based
+				require.Contains(t, term.Output(), "No Non-operator deployment found in namespace", tc.namespace, ", hence no restart happened")
+				assert.Equal(t, 1, csCalls)
+			} else if tc.name1 == "autoscaling-buffer" {
+				require.Contains(t, term.Output(), "Found only autoscaling-buffer deployment in namespace toolchain-member-operator , which is not required to be restarted")
+				assert.Equal(t, 1, csCalls)
+			}
 
-			// then
-			require.NoError(t, err)
-			AssertDeploymentHasReplicas(t, fakeClient, namespacedName, 3)
-			assert.Equal(t, 0, numberOfUpdateCalls)
-			assert.Contains(t, term.Output(), "ERROR: The given deployment 'wrong-deployment' wasn't found.")
-			assert.Contains(t, term.Output(), fmt.Sprintf("Existing deployments in toolchain-%s-operator namespace", clusterType))
-			assert.Contains(t, term.Output(), "cool-deployment")
 		})
 	}
 }
 
-func TestRestartDeploymentWithInsufficientPermissions(t *testing.T) {
-	// given
-	SetFileConfig(t, Host(NoToken()), Member(NoToken()))
-	for _, clusterName := range []string{"host", "member1"} {
-		// given
-		clusterType := configuration.Host
-		if clusterName != "host" {
-			clusterType = configuration.Member
-		}
-		namespace := fmt.Sprintf("toolchain-%s-operator", clusterType)
-		namespacedName := types.NamespacedName{
-			Namespace: namespace,
-			Name:      "cool-deployment",
-		}
-		deployment := newDeployment(namespacedName, 3)
-		newClient, fakeClient := NewFakeClients(t, deployment)
-		numberOfUpdateCalls := 0
-		fakeClient.MockUpdate = requireDeploymentBeingUpdated(t, fakeClient, namespacedName, 3, &numberOfUpdateCalls)
-		term := NewFakeTerminalWithResponse("Y")
+func TestRestartDeployment(t *testing.T) {
+	//given
+	testIOStreams := genericclioptions.NewTestIOStreamsDiscard()
+	SetFileConfig(t, Host())
+	toolchainCluster := NewToolchainCluster(ToolchainClusterName("host"))
+	hostDeployment := newDeployment(test.NamespacedName("toolchain-host-operator", "host-operator-controller-manager"), 1)
+	hostDeployment.Labels = make(map[string]string)
+	hostDeployment.Labels["kubesaw-control-plane"] = "kubesaw-controller-manager"
```

Comment:
another way to simplify the code
```suggestion
	hostDeployment.Labels = map[string]string{"kubesaw-control-plane": "kubesaw-controller-manager"}
```